mm2src/coins/z_coin.rs

```diff
@@ -1129,11 +1211,20 @@ impl MarketCoinOps for ZCoin {
     fn my_balance(&self) -> BalanceFut<CoinBalance> {
         let coin = self.clone();
         let fut = async move {
-            let sat = coin
+            let balance_sat = coin
                 .my_balance_sat()
                 .await
                 .mm_err(|e| BalanceError::WalletStorageError(e.to_string()))?;
-            Ok(CoinBalance::new(big_decimal_from_sat_unsigned(sat, coin.decimals())))
+            let balance = big_decimal_from_sat_unsigned(balance_sat, coin.decimals());
+            let change = coin.z_fields.change_tracker.sum_all_change_notes();
+
+            if balance < change {
+                return MmError::err(BalanceError::WalletStorageError(
+                    "change can't be greater than balance".to_owned(),
```

Comment:
I am not sure if I am missing something but `WalletStorageError` doesn't sound like a good fit for this error.